tools/fiwalk/src/hash_t.h

```diff
@@ -245,12 +245,12 @@ class hash_generator__:T { 			/* generates the hash */
 	hashed_bytes += bufsize;
     }
     void release(){			/* free allocated memory */
-	if(initialized){
```

Comment:
Yes, it's correct. The `md` and `mdctx` get allocated whether the object is initialized or not. The problem came from my thinking that initialized referred both to the object and to the hash machine.